logical_switch.go

```diff
@@ -331,6 +331,66 @@ func (odbi *ovndb) lsExtIdsDelImp(ls string, external_ids map[string]string) (*O
 	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
 }
 
+// lsAuxAddImp is for adding keys/vals either to external_ids or other_config record
+func (odbi *ovndb) lsAuxAddImp(ls string, auxConf map[string]string, auxCol string) (*OvnCommand, error) {
+	if (auxCol != "external_ids") && (auxCol != "other_config") {
+		return nil, fmt.Errorf("%s is neither other_config nor external_ids", auxCol)
+	}
+	if len(auxConf) == 0 {
+		return nil, fmt.Errorf("%s map is nil or empty", auxCol)
+	}
+
+	mutateSet, err := libovsdb.NewOvsMap(auxConf)
+	if err != nil {
+		return nil, err
+	}
+
+	var operations []libovsdb.Operation
+	row := make(OVNRow)
+	row["name"] = ls
+
+	mutation := libovsdb.NewMutation(auxCol, opInsert, mutateSet)
+	condition := libovsdb.NewCondition("name", "==", ls)
+	mutateOp := libovsdb.Operation{
+		Op:        opMutate,
+		Table:     TableLogicalSwitch,
+		Mutations: []interface{}{mutation},
+		Where:     []interface{}{condition},
+	}
+	operations = append(operations, mutateOp)
+	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
+}
+
+// lsAuxDelImp is for removing keys/vals from either external_ids or other_config records
+func (odbi *ovndb) lsAuxDelImp(ls string, auxConf map[string]string, auxCol string) (*OvnCommand, error) {
+	if (auxCol != "external_ids") && (auxCol != "other_config") {
+		return nil, fmt.Errorf("%s is neither other_config nor external_ids", auxCol)
+	}
+	if len(auxConf) == 0 {
+		return nil, fmt.Errorf("%s map is nil or empty", auxCol)
+	}
+
+	mutateSet, err := libovsdb.NewOvsMap(auxConf)
```

Comment:
@hzhou8 that'd be useful, IMHO - empty map means "delete" all k/v for update

Reply:
> empty map means "delete" all k/v

Not sure if this a good idea. The API name suggests "delete" the stuff in the map, so users may be easily think empty means delete nothing.

> key: "" means delete this k/v

Yes, this is what I proposed, but I thought about a corner case when we do allow empty string "" as a value, and now we only wants to delete the key iff the value is exactly "". Unfortunately there is no "nil" for string in Go. Another option is to use map[string]*string as the parameter type, but it would seem inconvenience for the normal use cases. Any other ideas?

Reply:
yeah, sending a nil/empty map (especially, nil) and removing all K/Vs would be too aggressive/dangerous. WRT empty string vs nil ptr - the latter seems ok to me. other options are more complex/less attractive (if we ignore the corner case with "" as value). nil ptr for string is the simplest, if the delete key semantics is useful, I'd go for it.

Reply:
ok, so we may implement the generic k/v del API using map[string]*string.